GrandCentralBoard/Widgets/WebsiteAnalytics/WebsiteAnalyticsWidget.swift

```diff
@@ -9,44 +9,99 @@ import GCBCore
 
 
 final class WebsiteAnalyticsWidget: WidgetControlling {
+    private enum State {
+        case Loading
+        case Data(reports: [PageViewsRowReport])
+        case NoData
+        case Error
+    }
 
     private let widgetView: TableWidgetView
-    private let widgetViewWrapper: UIView
-    let sources: [UpdatingSource]
+    private let mainView: UIView
 
-    var view: UIView {
-        return widgetViewWrapper
+    private var state: State = .Loading {
+        didSet {
+            changeFromState(oldValue, toState: state)
+        }
     }
 
     init(sources: [UpdatingSource]) {
         self.widgetView = TableWidgetView.fromNib()
         self.sources = sources
 
-        let viewModel = WidgetTemplateViewModel(title: "BLOGPOSTS", subtitle: "MOST POPULAR", contentView: widgetView)
+        let viewModel = WidgetTemplateViewModel(title: "BLOGPOSTS".localized.uppercaseString,
+                                                subtitle: "MOST POPULAR".localized.uppercaseString,
```

Comment:
could be `"Most Popular".localized.uppercaseString` (easier to read). And the same for other strings.